### ISSUE TITLE
Raiden hangs after trying to transfer funds to open channel without having any funds from that token

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/raiden#746]
## Problem Definition

On an open channel, trying to transfer funds to it with `0` balance on that token, for example, with a request like this one:
```
curl 'http://localhost:5001/api/1/transfers/0xf99dc75b8aa1e46d34b32f97dc7e9902c0d5ff26/0x000397dfd32afaae870e6b5fb44154fd43e43224' -H 'Content-Type: application/json' --data-binary '{"amount":5,"identifier":58}'
```
hangs indefinitely (`pending` request status on Chrome), and also any other request made to Raiden API hangs. Only solution found is restarting Raiden.
On Raiden console output, no exception is Raised, only the following log lines:
```
INFO:raiden.raiden_service  DIRECT CHANNEL 008135fa > 000397df is closed or has no funding                                                                                         
INFO:raiden.network.channelgraph    channel 008135fa - 000397df is closed or has zero funding, ignoring
```

## Desired behavior

Raiden should reply the request with a proper error code, maybe `403` or `404`, as request funds were not found.

## System Description
Raiden from current master, over Parity.

**Comment by andrevmatos** [comment]
This bug was identified on webUI testing.

**Comment by agatsoh** [comment]
I have faced this error before

**Comment by pcppcp** [comment]
This is a little bit complicated to do. As we can't expect that any node helds any amount of funds, because we never really have a complete network state, we can't check if transfer will make it through.

I'll add check for direct transfers, but the web app should still check the transfer's result and not rely on E200 as a confirmation.

**Comment by pcppcp** [comment]
Fixed in PR #829
